src/dir.proj

```diff
@@ -0,0 +1,292 @@
+      <CrossgenArgs>$(CrossgenArgs) -JITPath $(JitPath)</CrossgenArgs>
+    </PropertyGroup>
+
+    <!-- Form the dynamic path that would not collide if another instance of this is running. -->
+    <MakeDir Directories="$(TmpDir)" />
+
+    <Exec Command="$(CrossgenPath) -readytorun -in %(PlatformAssemblies.Identity) -out $(TmpDir)/%(PlatformAssemblies.Filename)%(PlatformAssemblies.Extension) $(CrossgenArgs)"
+          EnvironmentVariables="COMPlus_PartialNGen=0" />
+
+    <ItemGroup>
+      <ReadyFiles Include="$(TmpDir)\*" />
+    </ItemGroup>
+    
+    <Copy SourceFiles="@(ReadyFiles)" DestinationFolder="$(SharedFrameworkNameAndVersionRoot)" />
+
+  </Target>
+
+  <Target Name="CopyHostArtifactsToSharedFramework">
+    <ItemGroup>
+      <HostArtifacts Include="$(CoreHostLockedDir)dotnet$(ExeSuffix)" />
+      <HostArtifacts Include="$(CoreHostLockedDir)$(DotnetHostFxrBaseName)" />
+      <!-- Hostpolicy should be the latest and not the locked version as it is supposed to evolve for -->
+      <!-- the framework and has a tight coupling with coreclr's API in the framework. -->
+      <HostArtifacts Include="$(CoreHostOutputDir)\$(HostPolicyBaseName)" />
+    </ItemGroup>
+    <Copy SourceFiles="@(HostArtifacts)" DestinationFolder="$(SharedFrameworkNameAndVersionRoot)" />
+  </Target>
+
+  <Target Name="GenerateRuntimeGraph">
+    <PropertyGroup>
+      <runtimeGraphGeneratorName>RuntimeGraphGenerator</runtimeGraphGeneratorName>
+      <runtimeGraphGeneratorProject>$(ProjectDir)setuptools/independent/$(runtimeGraphGeneratorName)</runtimeGraphGeneratorProject>
+      <runtimeGraphGeneratorOutput>$(IntermediateOutputRootPath)setuptools/independent/$(runtimeGraphGeneratorName)</runtimeGraphGeneratorOutput>
+      <runtimeGraphGeneratorRuntime Condition="'$(OSGroup)'=='Windows_NT'">win</runtimeGraphGeneratorRuntime>
+      <runtimeGraphGeneratorRuntime Condition="'$(OSGroup)'=='OSX'">osx</runtimeGraphGeneratorRuntime>
+      <runtimeGraphGeneratorRuntime Condition="'$(runtimeGraphGeneratorRuntime)'==''">linux</runtimeGraphGeneratorRuntime>      
+    </PropertyGroup>
+
+    <Error Text="Could not determine rid graph generation runtime for platform $(OSGroup)"
+          Condition="$(runtimeGraphGeneratorRuntime) == ''" />
+
+    <Exec Command="$(DotnetToolCommand) publish --output $(runtimeGraphGeneratorOutput) $(runtimeGraphGeneratorProject) "
+          EnvironmentVariables="NUGET_PACKAGES=$(PackagesDir)" />
+
+    <PropertyGroup>
+      <RuntimeGraphGeneratorExe>$(runtimeGraphGeneratorOutput)/$(runtimeGraphGeneratorName)$(ExeSuffix)</RuntimeGraphGeneratorExe>
+      <DepsJson>$(SharedFrameworkNameAndVersionRoot)\$(SharedFrameworkName).deps.json</DepsJson>
+
+    </PropertyGroup>
+
+    <Exec Command="$(RuntimeGraphGeneratorExe) --project $(SharedFrameworkSourceRoot) --deps $(DepsJson) $(runtimeGraphGeneratorRuntime)"
+          EnvironmentVariables="NUGET_PACKAGES=$(PackagesDir)" />
+  </Target>
+
+  <Target Name="RestoreLockedCoreHost">
```

Comment:
Same question here:  Why not use a .csproj for this file instead of using old/deprecated project.json?